### PR TITLE
convos: 7.02 -> 8.05

### PR DESCRIPTION
**Pull request by zakame**
## Description of changes

https://github.com/convos-chat/convos/blob/v8.05/Changes#L3-L119

Needs quite a few PRs merged in first:

- https://github.com/NixOS/nixpkgs/pull/262089
- https://github.com/NixOS/nixpkgs/pull/262092
- https://github.com/NixOS/nixpkgs/pull/262093
- https://github.com/NixOS/nixpkgs/pull/262095
- https://github.com/NixOS/nixpkgs/pull/262097
- https://github.com/NixOS/nixpkgs/pull/262098
- https://github.com/NixOS/nixpkgs/pull/262100
- https://github.com/NixOS/nixpkgs/pull/262101
- https://github.com/NixOS/nixpkgs/pull/263937

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by stigtsp** [comment]
Welcome back! :) The dependency bump PRs might conflict with commits in:
- https://github.com/NixOS/nixpkgs/pull/261516

I'll see if I can cherry-pick these in to prevent issues

**Comment by stigtsp** [comment]
@ofborg test convos

**Comment by zakame** [comment]
@ofborg test convos

**Comment by zakame** [comment]
Will add a small `substituteInPlace` patch for the version - see https://github.com/convos-chat/convos/issues/858

**Comment by stigtsp** [comment]
> Will add a small `substituteInPlace` patch for the version - see [convos-chat/convos#858](https://github.com/convos-chat/convos/issues/858)

If this is fixed quickly upstream, should we maybe just wait for 7.17?

**Comment by zakame** [comment]
Yep I can just wait for 7.17 too 🙏

**Comment by zakame** [comment]
Updating for `convos-8.02` - Needs a new dep via https://github.com/NixOS/nixpkgs/pull/263937

**Comment by stigtsp** [comment]
7dd2af9913e0db37c2c888616f4e1f0f52348c99 is already in staging-next

**Comment by stigtsp** [comment]
8.05 is out tho :)

File: pkgs/applications/networking/irc/convos/default.nix [hunk 1]
```diff
@@ -67,16 +69,24 @@ perlPackages.buildPerlPackage rec {
       > t/00_nixpkgs_module_install.t
   '';
 
+  # (temp) bump the version - see https://github.com/convos-chat/convos/issues/858
+  preInstall = ''
+    substituteInPlace lib/Convos.pm \
+      --replace '7.15' '7.16'
+    substituteInPlace script/convos \
+      --replace '7.15' '7.16'
+  '';
```
Comment (stigtsp) [inline pkgs/applications/networking/irc/convos/default.nix:78]: Is this still needed for v8.x?

Comment (zakame) [reply pkgs/applications/networking/irc/convos/default.nix:78]: Yeah this can be dropped, thanks for reminding me! 👍

**Comment by zakame** [comment]
8.05 will need https://github.com/NixOS/nixpkgs/pull/268374/commits/c72cb9956404d8c476ebc148647a2e9c782aa980 (Mojolicious: 9.34 -> 9.35)

**Comment by stigtsp** [comment]
> 8.05 will need [c72cb99](https://github.com/NixOS/nixpkgs/commit/c72cb9956404d8c476ebc148647a2e9c782aa980) (Mojolicious: 9.34 -> 9.35)

Ah ok, you can cherry pick it in to this PR if you'd like?

**Comment by zakame** [comment]
Yep it won't run without the new Mojo:

```
[nix-shell:~/.cache/nixpkgs-review/rev-d72fb53fad3f4486956170e37720fbce614b2d48-1]$ results/convos/bin/convos daemon

CORE
  Name                           Required   Installed
  perl                           5.020      v5.38.0
  YAML::XS                       0.88       0.88
  Unicode::UTF8                  0.62       0.62
  Time::Piece                    1.20       1.3401
  Text::Markdown::Hoedown        1.03       1.03
  Syntax::Keyword::Try           0.29       0.29
  Parse::IRC                     1.22       1.22
  Mojolicious::Plugin::Syslog    0.06       0.06
  Mojolicious::Plugin::OpenAPI   5.09       5.09
  Mojolicious                    9.35                  You have version 9.34
  Module::Install                1.10       1.21
  LinkEmbedder                   1.20       1.20
  IRC::Utils                     0.12       0.12
  IO::Socket::SSL                2.009      2.083
  HTTP::AcceptLanguage           0.02       0.02
  Future::AsyncAwait             0.66       0.66
  File::ReadBackwards            1.06       1.06
  File::HomeDir                  1.00       1.006
  Crypt::Passphrase::Bcrypt      0.007      0.007
  Crypt::Passphrase::Argon2      0.009      0.009
  Crypt::Passphrase              0.016      0.016
  Convos                         8.05       8.05


It is not possible to start Convos at this point, since there
are some missing dependencies that need to be installed.

```

**Comment by zakame** [comment]
Result of `nixpkgs-review pr 262104` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>130 packages built:</summary>
  <ul>
    <li>abcde</li>
    <li>convos</li>
    <li>convos.devdoc</li>
    <li>get_iplayer</li>
    <li>get_iplayer.man</li>
    <li>lanraragi</li>
    <li>perl536Packages.CPANAudit</li>
    <li>perl536Packages.CPANAudit.devdoc</li>
    <li>perl536Packages.JSONValidator</li>
    <li>perl536Packages.JSONValidator.devdoc</li>
    <li>perl536Packages.LinkEmbedder</li>
    <li>perl536Packages.LinkEmbedder.devdoc</li>
    <li>perl536Packages.Minion</li>
    <li>perl536Packages.Minion.devdoc</li>
    <li>perl536Packages.MinionBackendRedis</li>
    <li>perl536Packages.MinionBackendRedis.devdoc</li>
    <li>perl536Packages.MinionBackendSQLite</li>
    <li>perl536Packages.MinionBackendSQLite.devdoc</li>
    <li>perl536Packages.MinionBackendmysql</li>
    <li>perl536Packages.MinionBackendmysql.devdoc</li>
    <li>perl536Packages.MojoIOLoopDelay</li>
    <li>perl536Packages.MojoIOLoopDelay.devdoc</li>
    <li>perl536Packages.MojoIOLoopForkCall</li>
    <li>perl536Packages.MojoIOLoopForkCall.devdoc</li>
    <li>perl536Packages.MojoJWT</li>
    <li>perl536Packages.MojoJWT.devdoc</li>
    <li>perl536Packages.MojoPg</li>
    <li>perl536Packages.MojoPg.devdoc</li>
    <li>perl536Packages.MojoRedis</li>
    <li>perl536Packages.MojoRedis.devdoc</li>
    <li>perl536Packages.MojoSAML</li>
    <li>perl536Packages.MojoSAML.devdoc</li>
    <li>perl536Packages.MojoSQLite</li>
    <li>perl536Packages.MojoSQLite.devdoc</li>
    <li>perl536Packages.MojoUserAgentCached</li>
    <li>perl536Packages.MojoUserAgentCached.devdoc</li>
    <li>perl536Packages.Mojolicious</li>
    <li>perl536Packages.Mojolicious.devdoc</li>
    <li>perl536Packages.MojoliciousPluginAssetPack</li>
    <li>perl536Packages.MojoliciousPluginAssetPack.devdoc</li>
    <li>perl536Packages.MojoliciousPluginGravatar</li>
    <li>perl536Packages.MojoliciousPluginGravatar.devdoc</li>
    <li>perl536Packages.MojoliciousPluginI18N</li>
    <li>perl536Packages.MojoliciousPluginI18N.devdoc</li>
    <li>perl536Packages.MojoliciousPluginMail</li>
    <li>perl536Packages.MojoliciousPluginMail.devdoc</li>
    <li>perl536Packages.MojoliciousPluginOpenAPI</li>
    <li>perl536Packages.MojoliciousPluginOpenAPI.devdoc</li>
    <li>perl536Packages.MojoliciousPluginRenderFile</li>
    <li>perl536Packages.MojoliciousPluginRenderFile.devdoc</li>
    <li>perl536Packages.MojoliciousPluginStatus</li>
    <li>perl536Packages.MojoliciousPluginStatus.devdoc</li>
    <li>perl536Packages.MojoliciousPluginSyslog</li>
    <li>perl536Packages.MojoliciousPluginSyslog.devdoc</li>
    <li>perl536Packages.MojoliciousPluginTemplateToolkit</li>
    <li>perl536Packages.MojoliciousPluginTemplateToolkit.devdoc</li>
    <li>perl536Packages.MojoliciousPluginTextExceptions</li>
    <li>perl536Packages.MojoliciousPluginTextExceptions.devdoc</li>
    <li>perl536Packages.MojoliciousPluginWebpack</li>
    <li>perl536Packages.MojoliciousPluginWebpack.devdoc</li>
    <li>perl536Packages.Mojomysql</li>
    <li>perl536Packages.Mojomysql.devdoc</li>
    <li>perl536Packages.MusicBrainz</li>
    <li>perl536Packages.MusicBrainz.devdoc</li>
    <li>perl536Packages.OpenAPIClient</li>
    <li>perl536Packages.OpenAPIClient.devdoc</li>
    <li>perl536Packages.Yancy</li>
    <li>perl536Packages.Yancy.devdoc</li>
    <li>perl538Packages.CPANAudit</li>
    <li>perl538Packages.CPANAudit.devdoc</li>
    <li>perl538Packages.JSONValidator</li>
    <li>perl538Packages.JSONValidator.devdoc</li>
    <li>perl538Packages.LinkEmbedder</li>
    <li>perl538Packages.LinkEmbedder.devdoc</li>
    <li>perl538Packages.Minion</li>
    <li>perl538Packages.Minion.devdoc</li>
    <li>perl538Packages.MinionBackendRedis</li>
    <li>perl538Packages.MinionBackendRedis.devdoc</li>
    <li>perl538Packages.MinionBackendSQLite</li>
    <li>perl538Packages.MinionBackendSQLite.devdoc</li>
    <li>perl538Packages.MinionBackendmysql</li>
    <li>perl538Packages.MinionBackendmysql.devdoc</li>
    <li>perl538Packages.MojoIOLoopDelay</li>
    <li>perl538Packages.MojoIOLoopDelay.devdoc</li>
    <li>perl538Packages.MojoIOLoopForkCall</li>
    <li>perl538Packages.MojoIOLoopForkCall.devdoc</li>
    <li>perl538Packages.MojoJWT</li>
    <li>perl538Packages.MojoJWT.devdoc</li>
    <li>perl538Packages.MojoPg</li>
    <li>perl538Packages.MojoPg.devdoc</li>
    <li>perl538Packages.MojoRedis</li>
    <li>perl538Packages.MojoRedis.devdoc</li>
    <li>perl538Packages.MojoSAML</li>
    <li>perl538Packages.MojoSAML.devdoc</li>
    <li>perl538Packages.MojoSQLite</li>
    <li>perl538Packages.MojoSQLite.devdoc</li>
    <li>perl538Packages.MojoUserAgentCached</li>
    <li>perl538Packages.MojoUserAgentCached.devdoc</li>
    <li>perl538Packages.Mojolicious</li>
    <li>perl538Packages.Mojolicious.devdoc</li>
    <li>perl538Packages.MojoliciousPluginAssetPack</li>
    <li>perl538Packages.MojoliciousPluginAssetPack.devdoc</li>
    <li>perl538Packages.MojoliciousPluginGravatar</li>
    <li>perl538Packages.MojoliciousPluginGravatar.devdoc</li>
    <li>perl538Packages.MojoliciousPluginI18N</li>
    <li>perl538Packages.MojoliciousPluginI18N.devdoc</li>
    <li>perl538Packages.MojoliciousPluginMail</li>
    <li>perl538Packages.MojoliciousPluginMail.devdoc</li>
    <li>perl538Packages.MojoliciousPluginOpenAPI</li>
    <li>perl538Packages.MojoliciousPluginOpenAPI.devdoc</li>
    <li>perl538Packages.MojoliciousPluginRenderFile</li>
    <li>perl538Packages.MojoliciousPluginRenderFile.devdoc</li>
    <li>perl538Packages.MojoliciousPluginStatus</li>
    <li>perl538Packages.MojoliciousPluginStatus.devdoc</li>
    <li>perl538Packages.MojoliciousPluginSyslog</li>
    <li>perl538Packages.MojoliciousPluginSyslog.devdoc</li>
    <li>perl538Packages.MojoliciousPluginTemplateToolkit</li>
    <li>perl538Packages.MojoliciousPluginTemplateToolkit.devdoc</li>
    <li>perl538Packages.MojoliciousPluginTextExceptions</li>
    <li>perl538Packages.MojoliciousPluginTextExceptions.devdoc</li>
    <li>perl538Packages.MojoliciousPluginWebpack</li>
    <li>perl538Packages.MojoliciousPluginWebpack.devdoc</li>
    <li>perl538Packages.Mojomysql</li>
    <li>perl538Packages.Mojomysql.devdoc</li>
    <li>perl538Packages.MusicBrainz</li>
    <li>perl538Packages.MusicBrainz.devdoc</li>
    <li>perl538Packages.OpenAPIClient</li>
    <li>perl538Packages.OpenAPIClient.devdoc</li>
    <li>perl538Packages.Yancy</li>
    <li>perl538Packages.Yancy.devdoc</li>
  </ul>
</details>

**Comment by stigtsp** [comment]
@ofborg test convos

**Comment by stigtsp** [comment]
The nixos test fails on checking the "application available at..." message. Maybe removing this check is ok:

```diff
diff --git a/nixos/tests/convos.nix b/nixos/tests/convos.nix
index 8fe5892da9e5..06d8d30fcb14 100644
--- a/nixos/tests/convos.nix
+++ b/nixos/tests/convos.nix
@@ -22,7 +22,6 @@ in
   testScript = ''
     machine.wait_for_unit("convos")
     machine.wait_for_open_port(${toString port})
-    machine.succeed("journalctl -u convos | grep -q 'application available at.*${toString port}'")
     machine.succeed("curl -f http://localhost:${toString port}/")
   '';
 })
```

**Comment by zakame** [comment]
@ofborg test convos

**Review by stigtsp — APPROVED** [review]
LGTM

**Comment by zakame** [comment]
Thanks @stigtsp! 🎉

**Pull request merged**